src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/RazorLSPTextViewConnectionListener.cs

```diff
@@ -69,11 +69,13 @@ internal class RazorLSPTextViewConnectionListener : ITextViewConnectionListener
         _editorFeatureDetector = editorFeatureDetector;
         _editorOptionsFactory = editorOptionsFactory;
         _editorSettingsManager = editorSettingsManager;
-        _textManager = (IVsTextManager4)serviceProvider.GetService(typeof(SVsTextManager));
-
-        Assumes.Present(_textManager);
     }
 
+    /// <summary>
+    /// Gets instance of <see cref="IVsTextManager4"/>. This accesses COM object and requires to be called on the UI thread.
+    /// </summary>
+    private IVsTextManager4 TextManager => _textManager ??= (IVsTextManager4)_serviceProvider.GetService(typeof(SVsTextManager));
```

Comment:
It's fair if you'd prefer somebody on the Razor team make this change, but should this class import a `JoinableTaskContext` and assert the main thread here?

Reply:
good suggestion, done!